packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -0,0 +1,255 @@
+/// pressed with at least [ForcePressGestureRecognizer.startPressure].
+typedef GestureForcePressStartCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] for when a pointer that has
+/// pressed with at least [ForcePressGestureRecognizer.peakPressure].
+typedef GestureForcePressPeakCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] during the frames
+/// after the triggering of a [ForcePressGestureRecognizer.onStart] callback.
+typedef GestureForcePressUpdateCallback = void Function(ForcePressDetails details);
+
+/// Signature for when the pointer that previously triggered a
+/// [ForcePressGestureRecognizer.onStart] callback is no longer in contact
+/// with the screen.
+typedef GestureForcePressEndCallback = void Function(ForcePressDetails details);
+
+/// Recognizes a force press on devices that have force sensors.
+///
+/// Only the force from a single pointer is used to invoke events. A tap
+/// recognizer will win against this recognizer on pointer up as long as the
+/// pointer has not pressed with a force greater than
+/// [ForcePressGestureRecognizer.startPressure]. A long press recognizer will
+/// win when the press down time exceeds the threshold time as long as the
+/// pointer's pressure was never greater than
+/// [ForcePressGestureRecognizer.startPressure] in that duration.
+///
+/// As of 11-21-18 only iPhone 6S devices and higher numbered devices have force
+/// touch, with the exception of the iPhone XR.
+///
+/// Reported pressure will always be in the range [0.0, 1.0], where 1.0 is
+/// maximum pressure and 0.0 is minimum pressure.
+class ForcePressGestureRecognizer extends OneSequenceGestureRecognizer {
+  /// Creates a force press gesture recognizer.
+  ///
+  /// The [startPressure] defaults to 0.4, and [peakPressure] defaults to 0.85
+  /// where a value of 0.0 is no pressure and a value of 1.0 is maximum pressure.
+  ForcePressGestureRecognizer({
+    this.startPressure = 0.4,
+    this.peakPressure = 0.85,
+    Object debugOwner,
+  }) : assert(startPressure != null),
+       assert(peakPressure != null),
+       assert(peakPressure > startPressure),
+       super(debugOwner: debugOwner);
+
+  /// A pointer is in contact with the screen and has just pressed with a force
+  /// exceeding the [startPressure]. Consequently, if there were other gesture
+  /// detectors, only the force press gesture will be detected and all others
+  /// will be rejected.
+  ///
+  /// The position of the pointer is provided in the callback's `details`
+  /// argument, which is a [ForcePressDetails] object.
+  GestureForcePressStartCallback onStart;
+
+  /// A pointer is in contact with the screen and is either moving on the plane
+  /// of the screen, pressing the screen with varying forces or both
+  /// simultaneously.
+  ///
+  /// This callback will be invoked for every frame after the invocation of
```

Comment:
This mentions frames but I don't think that's accurate. Won't it be sent for every pointer event, rather than every frame? I don't see any code that tries to ensure it's frame-aligned or frame-buffered.

Reply:
yup!